### PR TITLE
Add GitHub Actions

### PR DESCRIPTION
**Pull request by simi**
I have removed testing against ruby-head for now. It is easy to add (ruby versions is already array which can be extended), but ruby-head can't even finish bundle and it will result in failed builds.

If we would like to test against `ruby-head` still, I can later contribute weekly action testing that outside of pull requests and push events.

:information_source: ~~Some checks fails in forks and are skipped, you can see full result at https://github.com/RubyElders/rubygems.org/pull/1.~~

:information_source: ~~I have removed "Tested by" part of footer. If needed, somebody can revive it later with GitHub logo.~~

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    name: Rails tests (Ruby ${{ matrix.ruby_version }}, RubyGems ${{ matrix.rubygems_version }})
+    runs-on: ubuntu-18.04
+    env:
+      RUBYGEMS_VERSION: ${{ matrix.rubygems_version }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install and start services
+      run: |
+        ./script/install_toxiproxy.sh
+        docker-compose up -d
```
Comment (simi) [inline .github/workflows/test.yml:23]: Notice this is done as early as possible to get some time for Elasticsearch to start.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -203,10 +203,6 @@
             <%= t("layouts.application.footer.gems_served_by") %>
             <span class="t-hidden">Fastly</span>
           </a>
-          <a class="footer__sponsor" href="https://travis-ci.org/" target="_blank">
-            <%= t("layouts.application.footer.tested_by") %>
-            <span class="t-hidden">Travis CI</span>
-          </a>
```
Comment (sonalkr132) [inline app/views/layouts/application.html.erb:209]: I think we can do this in a separate PR. We need to update the sprite image in the background or else the order of links will get messed up.

File: Gemfile [hunk 3]
```diff
@@ -52,11 +52,13 @@ end
 group :development, :test do
   gem "m", "~> 1.5", require: false
   gem "pry-byebug"
-  gem "rubocop", require: false
-  gem "rubocop-rails", require: false
-  gem "rubocop-performance", require: false
-  gem "brakeman", require: false
   gem "toxiproxy", "~> 1.0.0"
+
+  # please update .github/workflows/lint.yml on versions update
```
Comment (sonalkr132) [inline Gemfile:57]: could it be possible to move following to a group and install only these with `--without`? I am assuming this change is being made because we don't want to run entire `bundle install` on lint jobs.

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          rubocop_version: 0.89.1
+          rubocop_extensions: rubocop-performance:1.7.1 rubocop-rails:2.5.2
+          github_token: ${{ secrets.github_token }}
+          reporter: github-pr-check
+  brakeman:
+    name: Brakeman
+    runs-on: ubuntu-18.04
+    # runs only in main repo, this check doesn't work in forks
+    if: github.event.pull_request.head.repo.full_name == github.repository
```
Comment (sonalkr132) [inline .github/workflows/lint.yml:25]: Only devmasx/brakeman-linter-action doesn't work on forks, right? I don't think we **have** to use it. Just the build failure on `brakeman` failing should be enough. Using this means brakeman will be skipped on all PRs.

Comment (simi) [reply .github/workflows/lint.yml:25]: I'll check this. I thought it doesn't work just other way around (if you open PR in custom forks).

File: Gemfile (hunk 3 above)
Comment (simi) [reply Gemfile:57]: Yes, I tried to avoid full bundle. But any idea how to do that? Gems are by default in "default" group if I remember well. Should I name all groups except that one?

File: .github/workflows/lint.yml (hunk 4 above)
Comment (simi) [reply .github/workflows/lint.yml:25]: Any way I can fall back to simple custom action for now. Those pre-used actions do report problems inline as checks annotations. That's the only benefit.

File: Gemfile (hunk 3 above)
Comment (simi) [reply Gemfile:57]: Anyway I think we can move to full bundle, since it will get cached to avoid this problem.

Comment (sonalkr132) [reply Gemfile:57]: I think bundle --without default,test,development,assets should work. but yeah, if it is cached, probably using complete bundle install should also not be an issue. 
I was just hoping we don't have to maintain the dependencies list at two places since dependabot updates these.

**Comment by colby-swandale** [comment]
Not. Cool. I really don't appreciate the work that I've been doing on/off for 12 months being taken over without the common courtesy of asking first or marking any attribution of exploratory work already done in #2564 as I was still working on the final PR to add this functionality. :unamused:  :unamused: :unamused:

**Comment by simi** [comment]
@colby-swandale I did this from scratch and I started after https://github.com/rubygems/rubygems.org/pull/2564 was closed recently without any info on that. I tried to contribute in there, but I got no response as well. Current CI is not usable anymore and I'm just trying to push things forward.

File: Gemfile (hunk 3 above)
Comment (simi) [reply Gemfile:57]: I have moved to full bundle for now. I'll check how complex would be to implement https://github.com/rubygems/rubygems/issues/4048 and potentially we can later opt-in only specified groups to get improved build time for linters.

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (simi) [reply app/views/layouts/application.html.erb:209]: :+1:

File: Gemfile (hunk 3 above)
Comment (sonalkr132) [reply Gemfile:57]: This means we can remove the comment right?

Comment (sonalkr132) [reply Gemfile:57]: we have disabled dependabot update on rubocop*, so version pinning is fine for them. However please remove it from brakeman, or else it won't be automatically updated.

**Review by sonalkr132 — APPROVED** [review]
This looks good to me :+1: I am not sure if we have creds set up for docker push from actions. I will confirm and merge.

File: .github/workflows/test.yml (hunk 1 above)
Comment (sonalkr132) [reply .github/workflows/test.yml:23]: On travis we had to setup chrome explicitly to make js tests work. I am not sure how is github actions working.. perhaps we should add https://github.com/marketplace/actions/setup-chromedriver anyway?

Comment (deivid-rodriguez) [reply .github/workflows/test.yml:23]: Github actiions provides this software by default: https://github.com/actions/virtual-environments/blob/4b03527e45271076bec1d4fdd4b4cfaaf3d3c4e7/images/linux/Ubuntu1804-README.md#browsers-and-drivers.

Comment (sonalkr132) [reply .github/workflows/test.yml:23]: nice. thank you for checking.

File: Gemfile (hunk 3 above)
Comment (simi) [reply Gemfile]: done

File: script/build_docker.sh [hunk 5]
```diff
@@ -47,4 +51,4 @@ fi
 
 echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin quay.io
```
Comment (sonalkr132) [inline script/build_docker.sh:52]: For docker_password and docker_username to work, we need to create env using DOCKER_PASSWORD and DOCKER_USERNAME secrets,  https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

File: .travis.yml [hunk 6]
```diff
@@ -1,55 +0,0 @@
-language: ruby
```
Comment (sonalkr132) [inline .travis.yml:1]: please remove this file from .dockerignore,

File: script/build_docker.sh (hunk 5 above)
Comment (simi) [reply script/build_docker.sh:52]: ok, I'll pass them to env in workflow

**Comment by sonalkr132** [comment]
Thank you for working on this :medal_sports:
